### DISCUSSION TITLE
Creating a user session using userId in server

### DISCUSSION THREAD

**Discussion by Aadim1** [supabase/supabase#30186] [Questions] [answered]
Is there  a way to create a user session in the server(maybe edge function or dedicated server) using just the userId. 

Firebase equivalent

```
const uid = 'some-uid';

getAuth()
  .createCustomToken(uid)
  ```

**Comment by GaryAustin1** [comment, marked as answer]
No.  CreateUser will always assign a random UUID.

You could use an auth hook to add another claim to the token for your own user id though:
https://supabase.com/docs/guides/database/postgres/custom-claims-and-role-based-access-control-rbac

You can also generate your own JWT's if you want and sign them with the JWT secret.  But these would not be users in the Auth system and you would have to set the Authorization header yourself.
